src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/ExpressionLambdaRewriter.cs

```diff
@@ -402,6 +402,20 @@ private BoundExpression VisitBinaryOperator(BinaryOperatorKind opKind, MethodSym
             }
         }
 
+        private BoundExpression PromoteEnumOperand(BoundExpression operand, BoundExpression loweredOperand, TypeSymbol promotedType, bool isChecked)
+        {
+            var literal = operand as BoundLiteral;
+            if (literal != null)
+            {
+                // for compat reasons enum literals are directly promoted into underlying values
+                return Visit(literal.Update(literal.ConstantValue, promotedType));
```

Comment:
Please do not visit newly constructed nodes in the lowering phase. Construct the required result, using existing helper methods if necessary.